Concurrency limit could failed with tenants

We have a customer that get the error below. 

Extracting table `concurrency_limit` show me that the database contains duplicate entry, one with tenant, and one without tenants : 


```
xxx.ingestion|particle-flat-ingestion,"{""flowId"": ""particle-flat-ingestion"", ""running"": 0, ""tenantId"": ""default"", ""namespace"": ""xxx.ingestion""}",default,xxx.ingestion,particle-flat-ingestion,0
default|xxx.ingestion|particle-flat-ingestion,"{""flowId"": ""particle-flat-ingestion"", ""running"": 1, ""tenantId"": ""default"", ""namespace"": ""xxx.ingestion""}",default,xxx.ingestion,particle-flat-ingestion,1

```

truncate the table `concurrency_limit` fixed that issue. 

2 reasons potentially: 
- the migration script don't do this table 
- we insert bad data without tenants 

```
20:09:23.162 ERROR jdbc-queue-Execution_0 .c.u.ThreadUncaughtExceptionHandlers Caught an exception in Thread[#55,jdbc-queue-Execution_0,5,main]. Shutting down.
java.util.concurrent.CompletionException: org.jooq.exception.TooManyRowsException: Cursor returned more than one result
        at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source)
        at java.base/java.util.concurrent.CompletableFuture.completeThrowable(Unknown Source)
        at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(Unknown Source)
        at io.micrometer.core.instrument.internal.TimedRunnable.run(TimedRunnable.java:49)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.base/java.lang.Thread.run(Unknown Source)
Caused by: org.jooq.exception.TooManyRowsException: Cursor returned more than one result
        at org.jooq_3.20.8.POSTGRES.debug(Unknown Source)
        at org.jooq.impl.Tools.fetchOne(Tools.java:2970)
        at org.jooq.impl.ResultQueryTrait.fetchOne(ResultQueryTrait.java:508)
        at io.kestra.jdbc.runner.AbstractJdbcConcurrencyLimitStorage.fetchOne(AbstractJdbcConcurrencyLimitStorage.java:135)
        at io.kestra.jdbc.runner.AbstractJdbcConcurrencyLimitStorage.lambda$countThenProcess$1(AbstractJdbcConcurrencyLimitStorage.java:40)
        at org.jooq.impl.DefaultDSLContext.lambda$transactionResult0$3(DefaultDSLContext.java:533)
        at org.jooq.impl.Tools$3$1.block(Tools.java:6393)
        at java.base/java.util.concurrent.ForkJoinPool.unmanagedBlock(Unknown Source)
... 
```


Links: 
- https://kestra-io.slack.com/archives/C07GM39T91V/p1763065006218769
- https://app.usepylon.com/issues/views/all-issues?conversationID=b8d30973-c651-4688-978c-82ae98857735


 <!-- Pylon-Issue-ID:b8d30973-c651-4688-978c-82ae98857735 -->

--- a/jdbc/src/main/java/io/kestra/jdbc/runner/AbstractJdbcConcurrencyLimitStorage.java
+++ b/jdbc/src/main/java/io/kestra/jdbc/runner/AbstractJdbcConcurrencyLimitStorage.java
@@ -22,10 +22,10 @@ public class AbstractJdbcConcurrencyLimitStorage extends AbstractJdbcRepository 
     }
 
     /**
-     * Fetch the concurrency limit counter then process the count using the consumer function.
-     * It locked the raw and is wrapped in a transaction so the consumer should use the provided dslContext for any database access.
+     * Fetch the concurrency limit counter, then process the count using the consumer function.
+     * It locked the raw and is wrapped in a transaction, so the consumer should use the provided dslContext for any database access.
      * <p>
-     * Note that to avoid a race when no concurrency limit counter exists, it first always try to insert a 0 counter.
+     * Note that to avoid a race when no concurrency limit counter exists, it first always tries to insert a 0 counter.
      */
     public ExecutionRunning countThenProcess(FlowInterface flow, BiFunction<DSLContext, ConcurrencyLimit, Pair<ExecutionRunning, ConcurrencyLimit>> consumer) {
         return this.jdbcRepository
@@ -97,7 +97,7 @@ public class AbstractJdbcConcurrencyLimitStorage extends AbstractJdbcRepository 
     }
 
     /**
-     * Returns all concurrency limit from the database
+     * Returns all concurrency limits from the database
      */
     public List<ConcurrencyLimit> find(String tenantId) {
         return this.jdbcRepository
@@ -132,8 +132,7 @@ public class AbstractJdbcConcurrencyLimitStorage extends AbstractJdbcRepository 
             .and(field("namespace").eq(flow.getNamespace()))
             .and(field("flow_id").eq(flow.getId()));
 
-        return Optional.ofNullable(select.forUpdate().fetchOne())
-            .map(record -> this.jdbcRepository.map(record));
+        return this.jdbcRepository.fetchOne(select.forUpdate());
     }
 
     private void update(DSLContext dslContext, ConcurrencyLimit concurrencyLimit) {

